gene2phenotype_project/gene2phenotype_app/views/publication.py

```diff
@@ -175,12 +217,105 @@ def post(self, request, stable_id):
                     context={"lgd": lgd, "user": user}
                 )
 
+                # Insert new publication
                 if serializer_class.is_valid():
                     serializer_class.save()
-                    response = Response({'message': 'Publication added to the G2P entry successfully.'}, status=status.HTTP_201_CREATED)
                 else:
                     response = Response({"errors": serializer_class.errors}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
+                # Add extra data linked to the publication
+                if "phenotypes" in publication:
+                    # Expected structure:
+                    #   { "phenotypes": [{ "accession": "HP:0003974", "publication": 1 }] }
+                    for pheno in publication["phenotypes"]:
+                        hpo_terms = pheno['hpo_terms']
+                        for hpo in hpo_terms:
+                            phenotype_data = {
+                                "accession": hpo["accession"],
+                                "publication": pheno["pmid"]
+                            }
+                            try:
+                                lgd_phenotype_serializer = LGDPhenotypeSerializer(
+                                    data = phenotype_data,
+                                    context = {'lgd': lgd}
+                                )
+                                # Validate the input data
+                                if lgd_phenotype_serializer.is_valid():
+                                    # save() is going to call create()
+                                    lgd_phenotype_serializer.save()
+                                else:
+                                    response = Response({"errors": lgd_phenotype_serializer.errors}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
itsn't better to return 400 bad request response?

Reply:
The error can be caused by something wrong in the serializer. It coud be something like "Invalid phenotype accession" or "Problem fetching the phenotype source". Both are more an internal error.

Reply:
But maybe handling it as bad request is better. 
I'm going to update.